qiskit_ibm_runtime/utils/estimator_pub_result.py

```diff
@@ -0,0 +1,83 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+========================================================================================
+EstimatorPubResult result classes (:mod:`qiskit_ibm_runtime.utils.estimator_pub_result`)
+========================================================================================
+
+.. autosummary::
+   :toctree: ../stubs/
+"""
+
+from __future__ import annotations
+
+from typing import Sequence
+import plotly.graph_objects as go
+
+from qiskit.primitives.containers import PubResult
+
+
+class EstimatorPubResult(PubResult):
```

Comment:
What is the workflow for a user that gets estimator results from running on a QPU?
Do they need to construct an `EstimatorPubResult` by themselves? Could you write a snippet that starts with `results = job.result()` so that we can look at it? 🙂

Reply:
Right now, that's what needs to be done. The intent is for an estimator job to return estimator pub results instead so the method can be called directly.

Reply:
👍 makes sense
Right now, `SamplerPubResult` and other result containers live in `qiskit.primitives.containers`. Do we want to add `EstimatorPubResult` here and port it later, or write it directly in qiskit? The latter seems more natural (no deprecations needed), I'm not sure when it'd become available to us though

Reply:
Per offline discussion, we're adding it in runtime as opposed to `qiskit.primitives.containers` as the methods rely on the metadata created by runtime types, e.g., `ZneOptions`, to work.